test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/Filtering/FilterOperatorTests.cs

```diff
@@ -0,0 +1,410 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using System.Web;
+using FluentAssertions;
+using Humanizer;
+using JsonApiDotNetCore.Configuration;
+using JsonApiDotNetCore.MongoDb;
+using JsonApiDotNetCore.Queries.Expressions;
+using JsonApiDotNetCore.Serialization.Objects;
+using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Driver;
+using Xunit;
+
+namespace JsonApiDotNetCoreMongoDbExampleTests.IntegrationTests.Filtering
+{
+    public sealed class FilterOperatorTests : IClassFixture<IntegrationTestContext<TestableStartup>>
+    {
+        private readonly IntegrationTestContext<TestableStartup> _testContext;
+
+        public FilterOperatorTests(IntegrationTestContext<TestableStartup> testContext)
+        {
+            _testContext = testContext;
+            
+            _testContext.RegisterResources(builder =>
+            {
+                builder.Add<FilterableResource, string>();
+            });
+            
+            _testContext.ConfigureServicesAfterStartup(services =>
+            {
+                services.AddResourceRepository<MongoDbRepository<FilterableResource>>();
+            });
+
+            var options = (JsonApiOptions) _testContext.Factory.Services.GetRequiredService<IJsonApiOptions>();
+            options.EnableLegacyFilterNotation = false;
+            options.SerializerSettings.DateFormatString = "yyyy-MM-dd";
```

Comment:
> Dunno if this is the only solution but it's the only way I could work around tests failing because of the timezone I'm at.

The tests are supposed to cover our important use cases. When such a test fails, it provides an opportunity to explore what is going on. Is the test wrong, is it hitting a bug, can it be implemented differently or is this supposed to fail?

Reply:
I explained myself poorly last time, I said it was because of the timezone I was at but it's not exactly that.

When I remove line 38 I fail the test `Can_filter_comparison_on_DateTime` because the date value that is returned by the API has the time included in it but the expected value does not. Here's an example of a failed assertion:

```
Expected responseDocument.ManyData[0].Attributes["someDateTime"] to be <2001-01-01>, but found <2001-01-01 03:00:00>.
```

Personally I don't think this is a bug, I don't remember how postgres handles dates and whether this can have anything to do with mongo storing dates differently but it might be something along those lines...